### PR TITLE
fix(projects): ensure linked projects are recomputed correctly

### PR DESCRIPTION
**Pull request by luytena**

### CODE REVIEW

File: addon/controllers/search-project.js [hunk 1]
```diff
@@ -26,14 +26,20 @@ export default class SearchProjectController extends Controller {
...
         localId: this.model.id,
       });
-      link.save();
-      this.router.transitionTo("project.index");
+      try {
+        await link.save();
+        this.constructionProject.all.perform(this.model.id);
+      } catch {
+        await link.deleteRecord();
+        // Ignore, trying to link project that is already linked.
```
Comment (velrest) [inline addon/controllers/search-project.js:40]: Maybe add notification that link already exists.

File: addon/routes/project.js [hunk 2]
```diff
@@ -4,4 +4,12 @@ export default class ProjectRoute extends Route {
   model() {
     return this.modelFor("application")?.id;
   }
+
```
Comment (velrest) [inline addon/routes/project.js:7]: Not needed anymore

File: addon/controllers/search-project.js (hunk 1 above)
Comment (luytena) [reply addon/controllers/search-project.js:39]: Done

File: addon/routes/project.js (hunk 2 above)
Comment (luytena) [reply addon/routes/project.js]: Removed

**Comment by luytena** [comment]
> For the comment about the `trackedFunction` you can cross-reference this issue: [NullVoxPopuli/ember-resources#340](https://github.com/NullVoxPopuli/ember-resources/issues/340)

Thanks, added the comment.

**Comment by luytena** [comment]
Offline-approved by @velrest

**Pull request merged**